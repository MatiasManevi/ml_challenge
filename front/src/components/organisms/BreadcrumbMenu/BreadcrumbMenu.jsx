import React from 'react';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';

import './BreadcrumbMenu.scss';
import { TextButton } from 'components/atoms';

const BreadcrumbMenu = ({ categories }) => {
	return (
		<div className="breadcrumb-menu">
			{categories.map((category, i) => {
				return (
					<React.Fragment key={category.id}>
						<TextButton text={category.name} />
						{i !== categories.length - 1 && <ChevronRightIcon fontSize="small" />}
					</React.Fragment>
				);
			})}
		</div>
	);
};

export default BreadcrumbMenu;
