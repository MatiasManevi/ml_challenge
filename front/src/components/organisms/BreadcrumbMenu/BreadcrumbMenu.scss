@import 'assets/scss/variables';

.breadcrumb-menu {
	display: flex;
	padding: $space-16 0;
	color: $color-neutral;
	justify-content: left;
	align-items: center;
}
