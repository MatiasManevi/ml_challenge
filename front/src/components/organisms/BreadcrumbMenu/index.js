export { default } from 'components/organisms/BreadcrumbMenu/BreadcrumbMenu';
