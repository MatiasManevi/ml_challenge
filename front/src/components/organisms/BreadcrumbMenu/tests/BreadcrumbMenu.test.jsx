import React from 'react';
import { shallow } from 'enzyme';

import BreadcrumbMenu from '../BreadcrumbMenu';
import categories from './categories.fixtures';

describe('BreadcrumbMenu', () => {
	it('should match snapshot', () => {
		const wrapper = shallow(<BreadcrumbMenu categories={categories} />);
		expect(wrapper).toMatchSnapshot();
	});
});
