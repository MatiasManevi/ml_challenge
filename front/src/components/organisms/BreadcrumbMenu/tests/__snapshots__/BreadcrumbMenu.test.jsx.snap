// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbMenu should match snapshot 1`] = `
<div
  className="breadcrumb-menu"
>
  <TextButton
    text="Electrónica, Audio y Video"
  />
  <Memo(ForwardRef(ChevronRightIcon))
    fontSize="small"
  />
  <TextButton
    text="Audio"
  />
  <Memo(ForwardRef(ChevronRightIcon))
    fontSize="small"
  />
  <TextButton
    text="Reproductores Portátiles"
  />
  <Memo(ForwardRef(ChevronRightIcon))
    fontSize="small"
  />
  <TextButton
    text="Accesorios"
  />
  <Memo(ForwardRef(ChevronRightIcon))
    fontSize="small"
  />
  <TextButton
    text="Parlantes Portátiles"
  />
</div>
`;
